Fix completion with one arg and add optional class objects to completion list

mpanarin opened the pull request:
1. Fixes the issue of 1 arg completion snippet.
Before the `$0` would be inserted literally, as insertTextFormat was missing.

2. Allows adding class objects to completion. 
When you want to call a classmethod or staticmethod, class name is completed with snippet as if we are instantiating it. This is super annoying, especially if init has a lot of args.

mpanarin committed:
Fix wrong insertTextFormat in completion with one arg

mpanarin committed:
Allow adding class objects to completion list

enabled by default, if snippets are enabled in jedi completion

goanpeca commented:
Awesome @mpanarin !

- Did a quick pass and looks good. Just one question should the `should_include_class_objects` option be `False` by default?
- And does jedi offers such configuration or is something we would be doing just on the PyLS side?

Thanks!

mpanarin commented:
@goanpeca 
1. I updated that part. I think the only people who would be annoyed by this features are the ones, who have snippets disabled. So I added a check for that and made `should_include_class_objects` enabled by default
2. No, I don't think so. We submit an issue or PR to jedi, but is it worth it?

goanpeca commented:
1 👍🏼
2 No need, just wondering about that.

Thanks!

benmezger commented:
Thanks for this. Been having issues with the annoying arg completion, just tried out your fork and seems to be working fine over here.

mpanarin commented:
@goanpeca @ccordoba12 
As this one is ready, can it be merged?
Thanks!

goanpeca reviewed (approved):
Thanks for the work @mpanarin !

It works as expected on my local tests @ccordoba12

ccordoba12 committed:
Merge branch 'develop' into fix_completion
